tests/testsuite/build_dir.rs

```diff
@@ -12,19 +12,22 @@ fn binary_with_debug() {
             ".cargo/config.toml",
             r#"
             [build]
-            target-dir = "build"
+            target-dir = "target"
+            build-dir = "build"
             "#,
         )
         .build();
 
-    p.cargo("build")
-        .masquerade_as_nightly_cargo(&[])
+    p.cargo("build -Z unstable-options -Z build-dir")
+        .masquerade_as_nightly_cargo(&["build-dir"])
         .enable_mac_dsym()
         .run();
 
     assert_build_dir(p.root().join("build"), "debug", true);
-    assert_executeable_exists(p.root().join("build/debug"), "foo", true);
-    assert!(!p.root().join("target").exists());
+    assert_build_dir(p.root().join("target"), "debug", false);
```

Comment:
In reading these tests, the intent of the assertion isn't too clear (asserting that the target dir is a build dir? what does `false` mean?).  I think it'd be better to have two separate functions.  If they truly must share an implementation, then they can just be wrappers on top of that.